tests/unit/materializers/test_base_materializer.py

```diff
@@ -11,3 +11,207 @@
 #  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
 #  or implied. See the License for the specific language governing
 #  permissions and limitations under the License.
+from contextlib import ExitStack as does_not_raise
+
+import pytest
+
+from zenml.exceptions import StepInterfaceError
+from zenml.materializers import BuiltInMaterializer
+from zenml.materializers.base_materializer import BaseMaterializer
+from zenml.steps import Output, step
+
+
+def test_call_step_with_missing_materializer_for_type():
```

Comment:
More of a philosophical question, but those existing tests are testing code that is inside `base_step.py` and not actually inside the materializer. (The actual base materializer class is essentially empty). Should we place the tests in the file corresponding to the code they test (would be `base_step.py` and `default_materializer_registry.py` in this case) or group them thematically like you did here?

Reply:
I was a bit hesitant about this as well. I felt like we were testing the connection between the steps and the materializers. But on second thought, I think you are right. We are testing the code within the `base_step.py` and the tests should be placed accordingly. I am on it!

Reply:
Changed the structure of the tests back.